ipaserver/install/cainstance.py

```diff
@@ -1506,6 +1510,73 @@ def uncomment_rewriterule():
                 logger.debug("Successfully updated CRL")
             api.Backend.ra.override_port = None
 
+    def setup_acme(self) -> bool:
+        """
+        Set up ACME service, if needed.
+
+        Return False if ACME service was already set up, otherwise True.
+
+        """
+
+        # ACME LDAP database schema will be added by ipa-server-upgrade.
+        # It is fine if this subroutine runs *before* the schema update,
+        # because we only create the container objects.
+
+        if os.path.isdir(os.path.join(paths.PKI_TOMCAT, 'acme')):
+            return False  # it seems ACME service is already set up
+
+        configure_acme_acls()
+
+        # create ACME agent group (if not exist already) and user
+        self.ensure_group(ACME_AGENT_GROUP, "ACME RA accounts")
+        acme_user = self.admin_user = f"acme-{self.fqdn}"
+        result = self.create_user(
+            uid=acme_user,
+            cn=acme_user,
+            sn=acme_user,
+            user_type='agentType',
+            groups=[ACME_AGENT_GROUP],
+            force=True,
+        )
+        if result is None:
+            raise RuntimeError("Failed to add ACME RA user")
```

Comment:
The ACME RA users are not cleaned up on IPA server uninstall. Perhaps server-del should clean these up.

Reply:
I don't think `server-del` can do it because the users are in the `o=ipaca` subtree.  But certainly `ipa-server-install --uninstall` can do it.  I'll add the behaviour.

Reply:
I'm going to punt on this.  It turns out to be non-trivial due to servers (including DS) getting stopped as an initial uninstall step.  I spent most of a day on it and I'm pulling the plug - the presense of the old entries is not a problem (including for new install with same hostname, which is handled).

Reply:
The uninstaller for DL1 calls remove_master_from_managed_topology() in ipaserver/install/server/install.py

The first thing this calls is server-del. This would be a good place to put it.

Reply:
@rcritten I'll have another look at it some time this week.  Thanks!

Reply:
@rcritten yeah, moving the invocation of the user delete behaviour to `uninstall_check` (near where `remove_master_from_managed_topology` is called) resolves the issues.  Latest push resolves it.  Done as a separate commit (currently at tip of branch) so you can review it separately.